### DISCUSSION TITLE
Gateway doesn't receive sensor data?

### DISCUSSION THREAD

**Discussion by rknse** [timmbogner/Farm-Data-Relay-System#212] [Q&A] [answered]
Hi!

I've downloaded FDRS and configured a simple system:
- LoRa sensor (D1 mini, RFM95/SX1276)
  #define READING_ID    0x10  
  #define GTWY_MAC      0x01

- LoRa gateway 0x01 (D1 mini, RFM95/SX1276)
  #define LORAG_ACT      sendSerial();
  #define LORA1_ACT      sendSerial();

- MQTT gateway 0x00 (not yet connected)

### The problem 
The LoRa Gateway doesn't receive any data from my sensor, but it looks like it receives data from other nearby sensors... 

### Gateway output:
```
    Initializing FDRS Gateway!
    Address: 1
[1] LORA_NEIGHBOR_1: 10
[1] LORA_NEIGHBOR_2: 11
    Debugging verbosity level: 2
    RadioLib initialization successful!
    LoRa Initialized. Frequency: 868.00  Bandwidth: 125.00  SF: 7  CR: 5  SyncWord: 18  Tx Power: 10dBm
[2] Incoming LoRa packet of 150 bytes not processed.
[2] Incoming LoRa packet of 63 bytes not processed.
[2] Incoming LoRa packet of 98 bytes not processed.
[2] Incoming LoRa packet of 82 bytes not processed.
[2] Incoming LoRa packet of 82 bytes not processed.
```

### Sensor output:
```
    Id: 16 - Type: 3 - Data loaded: 45.70
    Id: 16 - Type: 1 - Data loaded: 22.60
    Sending FDRS Packet!
    Lora DR Buffer Overflow!
    Lora DR Buffer Overflow!
[2] DR added to LoRa buffer. start: 11 end: 10
     LoRa sent.
h=45.7	t=22.6
     Delaying.
```
And just now I noticed that the sensor got "DR buffer overflow" after a while - but it still says "LoRa sent"?

I have tested the same setup with an ESPNOW_Sensor and an ESPNOW_Gateway and this works fine! Unfortunately, I can't use ESPNOW since the sensor would be out of range...  

What am I missing? Can someone please give me a hint where the problem could be?

**Comment by timmbogner** [comment]
Hey @rknse,
Thanks a lot for trying it out. It does look like something's going wrong within the system itself. I'll test this tonight and get it figured out!

**Reply by timmbogner** [reply]
Fyi, the packets "not processed" are noise that gets registered as a packet by the SX1276 chip. It's normal, and they are probably not related to the issue.

**Comment by aviateur17** [comment]
@rknse, could you post the sensor serial output soon after it's powered up for about a minute or so of output?   I assume you have LoRa ack's enabled?   Those other packets received are not the correct length and thus they are not processed.

**Reply by rknse** [reply]
@aviateur17, below is the sensor output after a few minutes. The "NO ACK!" in the output shows that sendFDRS() always returns FALSE.
    Id: 16 - Type: 3 - Data loaded: 43.80
    Id: 16 - Type: 1 - Data loaded: 23.00
data_count=2
    Sending FDRS Packet!
[2] DR added to LoRa buffer. start: 0 end: 56
     LoRa sent.
_NO ACK!_
h=43.8	t=23.0
     Delaying.
    Id: 16 - Type: 3 - Data loaded: 43.70
    Id: 16 - Type: 1 - Data loaded: 23.10
data_count=2
    Sending FDRS Packet!
[2] DR added to LoRa buffer. start: 0 end: 58
     LoRa sent.
_NO ACK!_
h=43.7	t=23.1
     Delaying.
    Id: 16 - Type: 3 - Data loaded: 44.30
    Id: 16 - Type: 1 - Data loaded: 23.10
data_count=2
    Sending FDRS Packet!
[2] DR added to LoRa buffer. start: 0 end: 60
     LoRa sent.
_NO ACK!_
h=44.3	t=23.1
     Delaying.
    Id: 16 - Type: 3 - Data loaded: 44.20
    Id: 16 - Type: 1 - Data loaded: 23.10
data_count=2
    Sending FDRS Packet!
[2] DR added to LoRa buffer. start: 0 end: 62
     LoRa sent.
_NO ACK!_
h=44.2	t=23.1
     Delaying.

**Comment by timmbogner** [comment, marked as answer]
Hi @rknse, 
Sorry for the delay, I just noticed I never got back to you about this. This was likely caused by a known issue which should now be fixed. Let me know if you still have issues after updating to the newest version of FDRS.

**Comment by rknse** [comment]
Hi Timm!
I noticed your changes and downloaded the code immrdiately! Everything is
working perfekt! I'm working on a battery powered node right now.

Regards

/Rolf
Rolf Karlsson
070 771 4580



Den sön 7 juli 2024 20:26Timm Bogner ***@***.***> skrev:
